### PR TITLE
feat(hatchery/vsphere): provisioning worker model v2

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: engine/hatchery/vsphere/client.go [hunk 1]
```diff
@@ -300,16 +300,32 @@ func (h *HatcheryVSphere) launchClientOp(ctx context.Context, vm *object.Virtual
...
+		// For model v2, we search the credentials from the VM Model name
+		for i := range h.Config.GuestCredentials {
+			if h.Config.GuestCredentials[i].ModelVMWare == model.GetVSphereImage() {
+				auth.Username = h.Config.GuestCredentials[i].Username
+				auth.Password = h.Config.GuestCredentials[i].Password
+				break
+			}
+		}
+	} else {
+		// For model v2, we search the credentials from the CDS Worker model name
```
Comment (sguiheux) [inline engine/hatchery/vsphere/client.go:314]:
```suggestion
		// For model v1, we search the credentials from the CDS Worker model name
```

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#21298.1 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21298.1 ■   
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ■

**Comment by ovh-cds** [comment]
CDS Report build-cds#21298.1 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21298.1 ■   
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ■

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21298.2 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by richardlt — APPROVED** [review]

**Pull request merged**